### PR TITLE
SelectorDatePickerRowFormer - make default date an optional instead of NSDate()

### PR DESCRIPTION
**Pull request by andrewloyola**
SelectorDatePickerRowFormer - make date an optional, add support for default text and date in the SelectorDatePickerFormableRow protocol

I needed a way to add placeholder text for this  Template if the date had never been set before, so i've changed self.date to be an optional, and added 2 protocol methods that return 2 things:

1) An Optional Default date, the demo has been modified to return NSDate() giving it the same functionality as before
2) A string to fall back to if the default date is not set and self.date is still nil.

### CODE REVIEW

File: Former/Cells/FormSelectorDatePickerCell.swift [hunk 1]
```diff
@@ -26,6 +26,14 @@ public class FormSelectorDatePickerCell: FormCell, SelectorDatePickerFormableRow
         return displayLabel
     }
     
+    public func formDefaultDisplayLabelText() -> String? {
+        return "Not Set"
```
Comment (ra1028) [inline Former/Cells/FormSelectorDatePickerCell.swift]: I think nil is better.

File: Former/Cells/FormSelectorDatePickerCell.swift [hunk 2]
```diff
@@ -26,6 +26,14 @@ public class FormSelectorDatePickerCell: FormCell, SelectorDatePickerFormableRow
...
+    }
+    
+    public func formDefaultDisplayDate() -> NSDate? {
+        return NSDate()
```
Comment (ra1028) [inline Former/Cells/FormSelectorDatePickerCell.swift]: Also, nil is better.

**Comment by ra1028** [comment]
@andrewloyola 
Sorry for the late :bow:
I wrote some comments.
Would you plz fix it?
Then I'll release new pods version.

**Comment by ZacharyKhan** [comment]
@ra1028 Can you take care of this last one?

**Comment by ra1028** [comment]
Thanks @ZacharyKhan !
I don't think this is such a great proposal.
But, I'm agree with you if you wants to merge this.

**Comment by ZacharyKhan** [comment]
Could you expand the solution for all inline pickers so there is a placeholder option for all inline picker types?

Also, as suggested by @ra1028, it would be better to make the placeholder value an optional and only set the placeholder if it is not nil.

**Comment by ZacharyKhan** [comment]
I think it would be useful if properly implemented 👍

**Comment by ra1028** [comment]
@ZacharyKhan OK, please merge this PR.

**Comment by walsht** [comment]
@andrewloyola / @ra1028  How do I implement this?  Currently the logic will set Date() for FormSelectorDatePickerCell.

**Comment by walsht** [comment]
@trieupenandcoffee Thank you.  But, I am using InlineDatePickerRowFormer and 
`Nil cannot be assigned to type 'Date'`
```

        return InlineDatePickerRowFormer<LabelCell>(instantiateType: .Nib(nibName: "LabelCell")) { [weak self] in
            $0.titleLabel.text = title.localized()
            $0.titleLabel.textColor = darkGreen
            $0.titleLabel.font = Font.subHeadline
            $0.displayLabel.textColor = .black
            $0.displayLabel.font =  Font.body
            $0.clearButton.isHidden = false
            $0.clearButton.tag = type.rawValue
            $0.clearButton.addTarget(self, action: #selector(self?.clearDate(_:)), for: .touchUpInside)
            }.configure {
                $0.displayEditingColor = darkGreen
                if let date = type.value() as? Date {
                    $0.date = date
                } else {
                    $0.date = nil
                }
            }.inlineCellSetup {
                $0.datePicker.datePickerMode = .date
                
            }.displayTextFromDate {
                return $0.format(with: .long, locale: Locale.current)
            }.onDateChanged { [weak self] in
                self?.updateCard(withRowType: type, withValue: $0)
        }
```

**Pull request merged**